src/main/java/httpserver/HttpServer.java

```diff
@@ -98,28 +102,32 @@ public void run() {
                     HttpRequest request = new HttpRequest(getRouter(), connection);
                     Thread t = new Thread(request);
                     t.start();
+                    
+					logger.info(String.format(
+							"Http request from %s:%d",
+							connection.getInetAddress(), connection.getPort()));
+                    
                 } catch (SocketException e) {
                     /*  This typically occurs when the client breaks the connection,
                         and isn't an issue on the server side, which means we shouldn't
                         break
                         */
-                    System.err.println("Client broke connection early!");
+                    	logger.warning("Client broke connection early!");
                     e.printStackTrace();
                 } catch (IOException e) {
                     /*  This typically means there's a problem in the HttpRequest
                     */
-                    System.err.println("IOException. Probably an HttpRequest issue.");
-                    e.printStackTrace();
+                    logger.log(Level.WARNING, "IOException. Probably an HttpRequest issue.", e);
+                    
                 } catch (HttpException e) {
-                    System.err.println("HttpException.");
-                    e.printStackTrace();
+                   
+                	logger.log(Level.WARNING, "HttpException.", e);
+                	
                 } catch (Exception e) {
                     /*  Some kind of unexpected exception occurred, something bad might
                         have happened.
                         */
-                    System.err.println("Generic Exception!");
-                    e.printStackTrace();
-
+                    logger.log(Level.SEVERE, "Generic Exception!", e);
```

Comment:
The newline here was for readability, because the error printing isn't related to the large comment that follows.